Model/Macro.cs

```diff
@@ -75,53 +75,44 @@ public string GetConfiguration()
-            if (roClient != null)
+            foreach (MacroConfig mc in this.configs)
             {
-                Thread macroThread = new Thread(() => {
-                    while (true)
+                if (mc.trigger != Key.None && Keyboard.IsKeyDown(mc.trigger))
+                {
+                    Dictionary<string, Key> macro = mc.macroEntries;
+                    for (int i = 1; i <= macro.Count; i++)//Ensure to execute keys in Order
                     {
-                        try
+                        Key macroKey = macro["in" + i + "mac" + mc.id];
+                        if (macroKey != Key.None)
                         {
-                            foreach(MacroConfig mc in this.configs)
-                            {
-                                if (mc.trigger != Key.None && Keyboard.IsKeyDown(mc.trigger))
-                                {
-                                    Dictionary<string,Key> macro = mc.macroEntries;
-                                    for (int i = 1; i <= macro.Count;i++)//Ensure to execute keys in Order
-                                    {
-                                        Key macroKey = macro["in" + i + "mac" + mc.id];
-                                        if(macroKey != Key.None)
-                                        {
-                                            Keys thisk = (Keys)Enum.Parse(typeof(Keys), macroKey.ToString());
-                                            Thread.Sleep(mc.delay);
-                                            Interop.PostMessage(roClient.process.MainWindowHandle, Constants.WM_KEYDOWN_MSG_ID, thisk, 0);
-                                        }
-
-                                    }
-                                }
-                            }
+                            Keys thisk = (Keys)Enum.Parse(typeof(Keys), macroKey.ToString());
+                            Thread.Sleep(mc.delay);
+                            Interop.PostMessage(roClient.process.MainWindowHandle, Constants.WM_KEYDOWN_MSG_ID, thisk, 0);
                         }
-                        catch (Exception) { }
-                        Thread.Sleep(100);
-                    }
-                });
 
-                this.macroThread = macroThread;
-                macroThread.SetApartmentState(ApartmentState.STA);
-                macroThread.Start();
+                    }
+                }
             }
+            Thread.Sleep(100);
+            return 0;
         }
 
-        public void Stop()
+        public void Start()
         {
-            if (this.macroThread != null && this.macroThread.IsAlive)
+            Stop();
+            Client roClient = ClientSingleton.GetClient();
+            if (roClient != null)
             {
-                this.macroThread.Abort();
+                this.thread = new _4RThread((_) => MacroExecutionThread(roClient));
+                _4RThread.Start(this.thread);
```

Comment:
```suggestion
                this._4RThread.Start(this.thread);
```

Reply:
this is not a instance attribute. 
`_4RThread` is a class with static `Start` method.